app/pages/user.js

```diff
@@ -0,0 +1,75 @@
+// user.js - profile page
+import { connect } from "react-redux";
+import Layout from "../components/Layout";
+import axios from "axios";
+import { API_URL } from "../utils/constants";
+import { useRouter } from "next/router";
+import ProfileBanner from "../components/atoms/ProfileBanner";
+import UserCard from "../components/molecules/UserCard";
+import styled from "styled-components";
+import ProfileNavigationBar from "../components/molecules/ProfileNavigationBar";
+import ProfileInformation from "../components/molecules/ProfileInformation";
+import ProfileStream from "../components/organisms/ProfileStream";
+import ProfileEvents from "../components/organisms/ProfileEvents";
+import ProfileTopics from "../components/organisms/ProfileTopics";
+import ProfileGroups from "../components/organisms/ProfileGroups";
+import React, { useState, useEffect } from "react";
+
+const ProfileContentContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  margin-top: 150px;
+`;
+
+const ProfileLeftColumn = styled.div``;
+
+const ProfileRightColumn = styled.div`
+  margin-top: 20px;
+`;
+
+function User({ name }) {
+  const router = useRouter();
+
+  const [activePanel, setActivePanel] = useState("stream");
```

Comment:
Warning untested code below:

```suggestion
  const [currentTab, setCurrentTab] = useState("stream");
  const tabs = getAvailableTabs();
  
  if (typeof tabs[currentTab] === "undefined") {
    throw new Error("Invalid tab was somehow selected, check business logic");
    } 
    
  const SelectedTab = tabs[currentTab];
```

This allows getAvailableTabs to make a network request because tabs could change with extensions.

For now it may just be something like
```
const getAvailableTabs = () => ({
  'stream' => ProfileStream,
  'events' => ProfileEvents,
 // etc.
});
```